library/src/main/java/com/google/android/exoplayer/MediaCodecUtil.java

```diff
@@ -95,34 +105,72 @@ public static synchronized void warmCodec(String mimeType, boolean secure) {
   /**
    * Returns the name of the best decoder and its capabilities for the given mimeType.
    *
-   * @param mimeType The mime type.
+   * @param originalMimeType The mime type.
    * @param secure Whether the decoder is required to support secure decryption. Always pass false
    *     unless secure decryption really is required.
    * @return The name of the best decoder and its capabilities for the given mimeType, or null if
    *     no decoder exists.
    */
   public static synchronized Pair<String, CodecCapabilities> getMediaCodecInfo(
-      String mimeType, boolean secure) throws DecoderQueryException {
-    CodecKey key = new CodecKey(mimeType, secure);
-    if (codecs.containsKey(key)) {
-      return codecs.get(key);
-    }
-    MediaCodecListCompat mediaCodecList = Util.SDK_INT >= 21
-        ? new MediaCodecListCompatV21(secure) : new MediaCodecListCompatV16();
-    Pair<String, CodecCapabilities> codecInfo = getMediaCodecInfo(key, mediaCodecList);
-    // TODO: Verify this cannot occur on v22, and change >= to == [Internal: b/18678462].
-    if (secure && codecInfo == null && Util.SDK_INT >= 21) {
-      // Some devices don't list secure decoders on API level 21. Try the legacy path.
-      mediaCodecList = new MediaCodecListCompatV16();
+      String originalMimeType, boolean secure) throws DecoderQueryException {
```

Comment:
The Sony Android TVs registered MediaCodecs against a different MimeType, at least at one point. I've received a few firmware updates, so it's possible that they have changed that. I think this issue is likely to happen on a percentage of devices which bundle their own DTS/DTS-HD/TrueHD decoder, as there isn't any concrete specifications to define exactly what is expected for these codecs. Native decoding for these codecs is still fairly rare, but new devices are coming on to the market which include these. I would imagine any future Android TV based TV devices would also fall into this set, unless these MimeTypes are being checked during CTP?